src/server/service/match.ts

```diff
@@ -325,6 +340,23 @@ export class Match {
             score: sql`${users.score} + ${player.score}`,
           })
           .where(eq(users.id, player.userId));
+
+        player.achievements
+          .filter((achievement) => {
+            return (
+              achievement === "beginnersLuck" ||
+              achievement === "realHuman" ||
+              achievement === "firstTimer" ||
+              achievement === "busterStreak"
+            );
+          })
+          .map(async (achievementId) => {
+            await dbTx.insert(userAchievements).values({
```

Comment:
You can update the users in just one single call if you pass an array of values instead. Prefer that. As an example, look at how we insert matches